app/controllers/admin/media_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Admin::MediaController < Admin::AdminController
+  before_action :find_attachable
+  before_action :find_media
+
+  def update
+    authorize @attachable
+    authorize @media
+    @media.update_attributes(media_params)
+
+    redirect_to [:admin, @attachable]
+  end
+  alias_method :create, :update
+
+  def destroy
+    authorize @attachable
+    authorize @media
+    @media.destroy
+
+    redirect_to [:admin, @attachable]
+  end
+
+  private
+
+  def find_media
+    @media = Media.find_or_initialize_by(id: params[:id], media_attachable: @attachable)
+  end
+
+  def find_attachable
+    @attachable = params[:media_attachable_type].classify.constantize.find(params[:media_attachable_id])
+  end
+
+  def media_params
+    params.require(:media).permit(:item, :caption_en, :caption_es, :'caption_es-AR', :caption_fr)
```

Comment:
I know the locale stuff is in flux but should be easy enough to build this list with a loop instead of hard coding?